src/xpk/core/blueprint/blueprint_generator.py

```diff
@@ -354,6 +360,289 @@ def _get_a3_mega_blueprint_dependencies(
+        settings={
+            "network_name": f"{cluster_name}-a3u-net-0",
+            "subnetworks": [{
+                "subnet_name": f"{cluster_name}-a3u-sub-0",
+                "subnet_region": region,
+                "subnet_ip": "192.168.0.0/18",
+            }],
+            "secondary_ranges": {
+                f"{cluster_name}-a3u-sub-0": [
+                    {"range_name": "pods", "ip_cidr_range": "10.4.0.0/14"},
+                    {"range_name": "services", "ip_cidr_range": "10.0.32.0/20"},
+                ]
+            },
+            "firewall_rules": [{
+                "name": f"{cluster_name}-a3u-internal-0",
+                "ranges": ["192.168.0.0/16"],
+                "allow": [
+                    {"protocol": "tcp", "ports": ["0-65535"]},
+                    {"protocol": "udp", "ports": ["0-65535"]},
+                    {"protocol": "icmp"},
+                ],
+            }],
+        },
+    )
+    net_1_id = f"{cluster_name}-a3u-net-1"
+    gpu_net_1 = DeploymentModule(
+        id=net_1_id,
+        source="github.com/GoogleCloudPlatform/cluster-toolkit.git//modules/network/vpc?ref=e0c690b",
+        settings={
+            "network_name": f"{cluster_name}-a3u-net-1",
+            "mtu": mtu_size,
+            "subnetworks": [{
+                "subnet_name": f"{cluster_name}-a3u-sub-1",
+                "subnet_region": region,
+                "subnet_ip": "192.168.64.0/18",
+            }],
+            "firewall_rules": [{
+                "name": f"{cluster_name}-a3u-internal-1",
+                "ranges": ["192.168.0.0/16"],
+                "allow": [
+                    {"protocol": "tcp", "ports": ["0-65535"]},
+                    {"protocol": "udp", "ports": ["0-65535"]},
+                    {"protocol": "icmp"},
+                ],
+            }],
+        },
+    )
+    rma_net_id = f"{cluster_name}-a3u-rdma-net"
+    rma_net = DeploymentModule(
+        id=rma_net_id,
+        source="github.com/GoogleCloudPlatform/cluster-toolkit.git//community/modules/network/rdma-vpc?ref=98c49fe",
+        settings={
+            "network_name": f"{cluster_name}-a3u-rdma-net",
+            "mtu": mtu_size,
+            "network_profile": f"https://www.googleapis.com/compute/beta/projects/{project_id}/global/networkProfiles/{zone}-vpc-roce",
+            "network_routing_mode": "REGIONAL",
+            "subnetworks_template": {
+                "name_prefix": f"{cluster_name}-a3u-rdma-sub",
+                "count": 8,
+                "ip_range": "192.168.128.0/18",
+                "region": region,
+            },
+        },
+    )
+    cluster_id = f"{cluster_name}-a3-ultragpu-cluster"
+    a3_ultra_cluster = DeploymentModule(
+        id=cluster_id,
+        source="github.com/GoogleCloudPlatform/cluster-toolkit.git//modules/scheduler/gke-cluster?ref=e0c690b",
+        use=[net_0_id],
+        settings={
+            "release_channel": "RAPID",
```

Comment:
Not sure if we need this anymore, please confirm with CT team and remove it so that we have a stable version of v1.31.
Before accepting any newer newer versions of GKE, we need to make sure they are compatible with a3 drivers. We'll discuss this when we implement --gke-version flag for A3 machines.

Reply:
confirmed that we need rapid release channel